kaldb/src/test/java/com/slack/kaldb/logstore/FieldConflictsTest.java

```diff
@@ -0,0 +1,227 @@
+    Tracing.newBuilder().build();
+  }
+
+  @Rule
+  public TemporaryLogStoreAndSearcherRule strictLogStore =
+      new TemporaryLogStoreAndSearcherRule(false);
+
+  public FieldConflictsTest() throws IOException {}
+
+  @Test
+  public void testFieldConflictingFieldTypeWithSameValue() throws InterruptedException {
+    final String conflictingFieldName = "conflictingField";
+
+    LogMessage msg1 =
+        new LogMessage(
+            MessageUtil.TEST_INDEX_NAME,
+            "INFO",
+            "1",
+            Map.of(
+                LogMessage.ReservedField.TIMESTAMP.fieldName,
+                MessageUtil.getCurrentLogDate(),
+                LogMessage.ReservedField.MESSAGE.fieldName,
+                "Test message",
+                LogMessage.ReservedField.TAG.fieldName,
+                "foo-bar",
+                LogMessage.ReservedField.HOSTNAME.fieldName,
+                "host1-dc2.abc.com",
+                conflictingFieldName,
+                "1"));
+    strictLogStore.logStore.addMessage(msg1);
+
+    LogMessage msg2 =
+        new LogMessage(
+            MessageUtil.TEST_INDEX_NAME,
+            "INFO",
+            "2",
+            Map.of(
+                LogMessage.ReservedField.TIMESTAMP.fieldName,
+                MessageUtil.getCurrentLogDate(),
+                LogMessage.ReservedField.MESSAGE.fieldName,
+                "Test message",
+                LogMessage.ReservedField.TAG.fieldName,
+                "foo-bar",
+                LogMessage.ReservedField.HOSTNAME.fieldName,
+                "host1-dc2.abc.com",
+                conflictingFieldName,
+                1));
+    strictLogStore.logStore.addMessage(msg2);
+
+    strictLogStore.logStore.commit();
+    strictLogStore.logStore.refresh();
+    Thread.sleep(1000);
+
+    final String queryByHost = "hostname:host1-dc2.abc.com";
+    Collection<LogMessage> resultsByHost =
+        findAllMessages(
+            strictLogStore.logSearcher, MessageUtil.TEST_INDEX_NAME, queryByHost, 1000, 1);
+    assertThat(resultsByHost.size()).isEqualTo(2);
+
+    final String conflictingTypeByNumber = conflictingFieldName + ":1";
+    Collection<LogMessage> searchByInt =
+        findAllMessages(
+            strictLogStore.logSearcher,
+            MessageUtil.TEST_INDEX_NAME,
+            conflictingTypeByNumber,
+            1000,
+            1);
+    assertThat(searchByInt.size()).isEqualTo(2);
+
+    final String conflictingTypeExactMatch = conflictingFieldName + ":\"1\"";
+    Collection<LogMessage> searchByNumber =
+        findAllMessages(
+            strictLogStore.logSearcher,
+            MessageUtil.TEST_INDEX_NAME,
+            conflictingTypeExactMatch,
+            1000,
+            1);
+    assertThat(searchByNumber.size()).isEqualTo(2);
+  }
+
+  @Test
+  public void testFieldConflictingFieldTypeWithDifferentValue() throws InterruptedException {
+    final String conflictingFieldName = "conflictingField";
```

Comment:
We should use a different conflicting field names for each test. We don't reset the log store so there's a chance there's shared state across each of these tests. Asserts that are in the subsequent test can pass due to state initialized in the prior test.

Reply:
The @Rule annotation resets the store for every test run. Further, we check the total number of messages for each test, and we can see that the unit tests are cleared.

Reply:
Ah, I misread this as a ClassRule. Carry on. 👍